internal/stacks/stackruntime/internal/stackeval/for_each.go

```diff
@@ -69,7 +69,20 @@ func evaluateForEachExpr(ctx context.Context, expr hcl.Expression, phase EvalPha
 	case ty.IsObjectType() || ty.IsMapType():
 		// okay
 
+	// TODO: Should we just remove this case? It seems like we can do some validation also on unknown values
```

Comment:
Note that the cases are checked in order, so if it is an object type or map type it already triggered the branch above and won't go into this one even if the value is unknown.

With this change we'll error if the value is `cty.UnknownVal(cty.DynamicType)` which is probably not what we want. So maybe a better check would be to move this below the ty.IsSetType() check below, and have it be `!result.Value().IsKnown() && ty == cty.DynamicPseudoType` so that we'll pass for that as well?